modules/nvidia-gpu-gcp-adding-a-gpu-node.adoc

```diff
@@ -0,0 +1,285 @@
+----
++
+.Example output
++
+[source,terminal]
+----
+NAME                                        DESIRED   CURRENT   READY   AVAILABLE   AGE
+preserve-dsoc12r4-ktjfc-worker-us-east-2a   1         1         1       1           3d11h
+preserve-dsoc12r4-ktjfc-worker-us-east-2b   2         2         2       2           3d11h
+----
+
+. View the machines that exist in the `openshift-machine-api` namespace by running the following command. At this time, there is only one compute machine per machine set, though a compute machine set could be scaled to add a node in a particular region and zone.
++
+[source,terminal]
+----
+$ oc get machines -n openshift-machine-api | grep worker
+----
++
+.Example output
++
+[source,terminal]
+----
+preserve-dsoc12r4-ktjfc-worker-us-east-2a-dts8r      Running   m5.xlarge   us-east-2   us-east-2a   3d11h
+preserve-dsoc12r4-ktjfc-worker-us-east-2b-dkv7w      Running   m5.xlarge   us-east-2   us-east-2b   3d11h
+preserve-dsoc12r4-ktjfc-worker-us-east-2b-k58cw      Running   m5.xlarge   us-east-2   us-east-2b   3d11h
+----
+
+. Make a copy of one of the existing compute `MachineSet` definitions and output the result to a JSON file by running the following command. This will be the basis for the GPU-enabled compute machine set definition.
++
+[source,terminal]
+----
+$ oc get machineset preserve-dsoc12r4-ktjfc-worker-us-east-2a -n openshift-machine-api -o json > <output_file.json>
+----
+
+. Edit the JSON file and make the following changes to the new `MachineSet` definition:
++
+* Replace `worker` with `gpu`. This will be the name of the new machine set.
+* Change the instance type of the new `MachineSet` definition to `g4dn`, which includes an NVIDIA Tesla T4 GPU.
+To learn more about AWS `g4dn` instance types, see link:https://aws.amazon.com/ec2/instance-types/#Accelerated_Computing[Accelerated Computing].
++
+[source,terminal]
+----
+$ jq .spec.template.spec.providerSpec.value.instanceType preserve-dsoc12r4-ktjfc-worker-gpu-us-east-2a.json
+
+"g4dn.xlarge"
+----
++
+The `<output_file.json>` file is saved as `preserve-dsoc12r4-ktjfc-worker-gpu-us-east-2a.json`.
+
+ . Update the following fields in `preserve-dsoc12r4-ktjfc-worker-gpu-us-east-2a.json`:
++
+* `.metadata.name` to a name containing `gpu`.
+
+* `.spec.selector.matchLabels["machine.openshift.io/cluster-api-machineset"]` to
+match the new `.metadata.name`.
+
+* `.spec.template.metadata.labels["machine.openshift.io/cluster-api-machineset"]`
+to match the new `.metadata.name`.
+
+* `.spec.template.spec.providerSpec.value.instanceType` to `g4dn.xlarge`.
+
+. To verify your changes, perform a `diff` of the original compute definition and the new GPU-enabled node definition by running the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-machine-api get preserve-dsoc12r4-ktjfc-worker-us-east-2a -o json | diff preserve-dsoc12r4-ktjfc-worker-gpu-us-east-2a.json -
+----
++
+.Example output
++
+[source,terminal]
+----
```

Comment:
Replace line 213 to 237 with this diff output:
```
15c15
<         "name": "myclustername-2pt9p-worker-gpu-a",
---
>         "name": "myclustername-2pt9p-worker-a",
25c25
<                 "machine.openshift.io/cluster-api-machineset": "myclustername-2pt9p-worker-gpu-a"
---
>                 "machine.openshift.io/cluster-api-machineset": "myclustername-2pt9p-worker-a"
34c34
<                     "machine.openshift.io/cluster-api-machineset": "myclustername-2pt9p-worker-gpu-a"
---
>                     "machine.openshift.io/cluster-api-machineset": "myclustername-2pt9p-worker-a"
59,60c59
<                         "machineType": "a2-highgpu-1g",
<                         "onHostMaintenance": "Terminate",
---
>                         "machineType": "n2-standard-4",
```

Reply:
changed.